### DISCUSSION TITLE
Unable to load dynamic library 'mysqli'

### DISCUSSION THREAD

**Discussion by tomaszbadon** [shivammathur/homebrew-php#73] [Q&A / Help] [answered]
Hi there,

I installed PHP 7.4 on my Mac using home brew using a tutorial:

https://getgrav.org/blog/macos-bigsur-apache-multiple-php-versions

and I enabled mysql in php.ini

The problem is that if I type php in the command line I receive:

`PHP Warning:  PHP Startup: Unable to load dynamic library 'mysqli' (tried: /usr/local/lib/php/pecl/20190902/mysqli (dlopen(/usr/local/lib/php/pecl/20190902/mysqli, 9): image not found), /usr/local/lib/php/pecl/20190902/mysqli.so (dlopen(/usr/local/lib/php/pecl/20190902/mysqli.so, 9): image not found)) in Unknown on line 0`

It's quite strange because I configured this particular version with apache2 in http.conf and I am able to use this mysql library.

`phpinfo();` displays also that mysqli is available.

If I run `php -a` it allows me to execute: `$link = mysqli_connect("localhost:3306", "user1", "password", "test");`.

I'd appreciate you help.

Thank you in advance.

BR,

Tomasz

**Comment by shivammathur** [comment, marked as answer]
To fix this run this command.
```zsh
php --ini
```

It will print the path to loaded configuration file and additional .ini files.
For example:
```zsh
Configuration File (php.ini) Path: /usr/local/etc/php/7.4
Loaded Configuration File:         /usr/local/etc/php/7.4/php.ini
Scan for additional .ini files in: /usr/local/etc/php/7.4/conf.d
Additional .ini files parsed:      /usr/local/etc/php/7.4/conf.d/ext-opcache.ini
```
check these files, if they have a line `extension=mysqli`, please remove that line as both `mysqli` and `mysqlnd` extensions are compiled statically into php binary and do not need to be loaded using a ini file.

If you are not able to find any such line, remove the PHP 7.4 along with its configuration and install it again.

```zsh
brew remove php@7.4
sudo rm -rf $(brew --prefix)/etc/php/7.4

brew update
brew install shivammathur/php/php@7.4
brew link --force --overwrite php@7.4
```

Restart your terminal, and test php version `php -v`

**Comment by tomaszbadon** [comment]
Thank you.

It works perfectly.

BR,

Tomasz
